app/javascript/components/remove-generic-item-modal.jsx

```diff
@@ -29,19 +30,26 @@ export const removeItems = (items, apiUrl, asyncDelete, redirectUrl) => {
         miqSparkleOff();
       } else {
         apiData.forEach(item => {
+          let flash = {};
           if (item.result === 'success') {
-            miqFlashLater({message: sprintf(deleteMessage, item.name)});
+            flash = {message: sprintf(deleteMessage, item.name), level: 'success'};
           } else if (item.result === 'error' && items.length > 1) {
-            miqFlashLater({message: sprintf(__('Error deleting item "%s": %s'), item.name, parseApiError(item.data)), level: 'error'});
+            flash = {message: sprintf(__('Error deleting item "%s": %s'), item.name, parseApiError(item.data)), level: 'error'};
           }
+          miqFlashLater(flash);
         });
       }
       return apiData;
     })
     .then((apiData) => {
       if (items.length > 1 || (items.length === 1 && apiData[0].result === 'success')) {
-        window.location.href = redirectUrl; // '/catalog/explorer?report_deleted=true';
-        miqSparkleOff();
+        if (typeof(treeSelect) == undefined) {
```

Comment:
Ah, another case of `typeof == undefined`. Fixing in https://github.com/ManageIQ/manageiq-ui-classic/pull/6984